### PR TITLE
"Microsoft Exchange On-Premise" integration breaks Data collection tasks since version 2.0.12

### PR DESCRIPTION
**Pull request by kobymeir**
## Status
- [x] Ready

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-32862

## Description
https://jira-dc.paloaltonetworks.com/browse/XSUP-32862
"Microsoft Exchange On-Premise" integration breaks Data collection tasks since version 2.0.12

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py [hunk 1]
```diff
@@ -752,6 +752,23 @@ def send_email_to_mailbox(account, to, subject, body, body_type, bcc, cc, reply_
...
+        body_type: type of the body. Can be 'html' or 'text' or None.
+        body: plain text body.
+        html_body: HTML formatted content (body) of the email to be sent.
+
+    Returns:
+        Body: the body of the message.
+    """
+    if body_type is None:
+        return HTMLBody(html_body) if html_body else Body(body)
+    return HTMLBody(html_body) if body_type.lower() == 'html' and html_body else Body(body)
```
Comment (dorschw) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py:769]: for readability, let's make it 
```suggestion
    if body_type.lower() == 'html' and html_body:
        return HTMLBody(html_body) 
    return Body(body)
```

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py [hunk 2]
```diff
@@ -752,6 +752,23 @@ def send_email_to_mailbox(account, to, subject, body, body_type, bcc, cc, reply_
...
+    Compatibility with 'send-mail' command - where body_type should be provided, we will use the body_type to decide.
+    Args:
+        body_type: type of the body. Can be 'html' or 'text' or None.
+        body: plain text body.
+        html_body: HTML formatted content (body) of the email to be sent.
+
+    Returns:
+        Body: the body of the message.
+    """
+    if body_type is None:
```
Comment (dorschw) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py:767]:
```suggestion
    if body_type is None: # when called from a data collection task
```

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2_test.py [hunk 3]
```diff
@@ -664,3 +664,48 @@ def test_get_time_zone(mocker):
     mocker.patch.object(demisto, 'callingContext', new={'context': {'User': {'timeZone': 'Asia/Jerusalem'}}})
     results = get_time_zone()
     assert results.key == 'Asia/Jerusalem'
+
+
+def test_get_message_for_body_type_no_body_type_with_html_body():
+    body = "This is a plain text body"
+    html_body = "<p>This is an HTML body</p>"
+    result = get_message_for_body_type(body, None, html_body)
```
Comment (dorschw) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2_test.py:672]: isn't it worth a `pytest.mark.parametrize`?

Comment (kobymeir) [reply Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2_test.py:672]: Anar doesn't like parameterize. so nope :)

**Review by dorschw — APPROVED** [review]

**Comment by kobymeir** [comment]
force merge - fails on unrelated test playbook issue

**Pull request merged**